doc/TockBinaryFormat.md

```diff
@@ -293,6 +309,44 @@ a non-position-independent binary at an incorrect location.
     the linker. If a fixed address is not required this should be set to
     `0xFFFFFFFF`.
 
+#### `6` Permissions
+
+The `Permissions` section allows an app to specify driver permissions that it
+is allowed to use. All driver syscalls that an app will use must be listed. The
+list should not include drivers that are not being used by the app.
+
+The data is stored in the optional `TbfHeaderV2Permissions` field. This
+includes an array of all the `perms`.
+
+```
+0             2             4
++-------------+-------------+---------...--+
+| Type (6)    | Length      | perms        |
++-------------+-------------+---------...--+
+```
+
+The `perms` array is made up of a number of elements of
+`TbfHeaderDriverPermission`. The length of the TLV can be used to determine
+the number of array elements. The elements in `TbfHeaderDriverPermission` are
+described below:
+
+`driver_number` is the number of the driver that is allowed. This for example
+could be `0x00000` to indicate that the `Alarm` syscalls are allowed.
+
+`allowed_commands` is a bit mask of the allowed commands. For example a value
+of `0b0001` indicates that only command 0 is allowed. `0b0111` would indicate
+that commands 2, 1 and 0 are all allowed. Subscribe and allow commands are
+always allowed as long as the specific `driver_number` has been specified. If
+a `driver_number` has not been specified for the capsule driver then `allow`
+and `subscribe` will be blocked.
+
+`driver_number` can be repeated. Each time `driver_number` is repeated the
```

Comment:
I worry about stateful specifications like this. It places requirements on a parser to maintain that state. 

Instead, and I think this came up in the call, there should be an offset field that specifies the offset of the bits. So you can split this into 2 32-bit values, and offset and a bitmask. The value of a bit in the bitmask is `bit_position` + `offset`. I.e., if bit 7 is set and the offset value is 300, it means command 307 is allowed.

This means that dense command spaces are more efficient to permission, but you can permission arbitrarily sparse ones not too badly (unlike say with this next-64 approach, which would not like the 9dof identifiers).

It would be good to also explicitly specify the layout of the the `TbfHeaderdriverPermission` structure as well.

Reply:
I was hesitant to do it that way as it adds an extra 4 bytes to every driver specified, when I think it will be rare that there will be more then 64 commands (assuming we change the current drivers to all use dense numbering).

I have converted it to using the `offset` field though.